### PR TITLE
fix: Cli not working with self hosted gitlab

### PR DESCRIPTION
**Pull request by jdalrymple**
fix: Fixing the inability to use environment variables in the cli export

chore: Cleaning up the rollup configuration
chore: Updating packages to remove vulnerabilities
docs: Updating usage documentation to improve clarity
test: Adding proper tests for cli functionality

BREAKING: Switch the version option to be a number and not a string. For example 4 instead of 'v4'.

closes:#458

### CODE REVIEW

**Comment by jdalrymple** [comment]
This could use a solid review if anyone has time :D

File: src/core/infrastructure/BaseService.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export interface BaseServiceOptions {
   jobToken?: string;
   host?: string;
   url?: string;
-  version?: 'v3' | 'v4';
+  version?: 3 | 4;
```
Comment (casz) [inline src/core/infrastructure/BaseService.ts]: This seems breaking 🤔

Comment (jdalrymple) [reply src/core/infrastructure/BaseService.ts]: Yea i mentioned it in the commit messages!

Comment (casz) [reply src/core/infrastructure/BaseService.ts]: yet major version wasn't bumped 😆

Comment (casz) [reply src/core/infrastructure/BaseService.ts]: Should have used rebase and merge

Comment (jdalrymple) [reply src/core/infrastructure/BaseService.ts]: yea im fixing it now

**Pull request merged**